### PR TITLE
Files are not removed if more than 10 to be removed

### PR DESCRIPTION
**Pull request by rfelker**
Missing key annotations, the post request body for searching filenames in elasticsearch was not created.  

The result is that too much data will be send back (because of the missing fields param) and only max 10 results will be responded (missing size param, default is 10). That's the reason why deleting files in elasticsearch doesn't work for folders with more than 10 files ("remove_deleted" : true).

Request size changed from 50000 to 10000. Values above 10000 will only work with pagination (50000 results in an exception and a fallback to 10).

Wrong key mapping for field fields in response

### CODE REVIEW

File: src/main/java/fr/pilato/elasticsearch/crawler/fs/client/SearchRequest.java [hunk 1]
```diff
@@ -19,11 +19,18 @@
 
 package fr.pilato.elasticsearch.crawler.fs.client;
 
+import com.google.api.client.json.GenericJson;
+import com.google.api.client.util.Key;
+
 import java.util.Arrays;
 
-public class SearchRequest {
+public class SearchRequest  extends GenericJson {
```
Comment (dadoonet) [inline src/main/java/fr/pilato/elasticsearch/crawler/fs/client/SearchRequest.java]: Could you explain why this change is needed?

File: src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java [hunk 2]
```diff
@@ -397,35 +399,47 @@ private void addFilesRecursively(FileAbstractor path, String filepath, Instant l
                     fsSettings.getElasticsearch().getIndex(),
                     fsSettings.getElasticsearch().getType(),
                     PATH_ENCODED + ":" + SignTool.sign(path),
-                    50000, // TODO: WHAT? DID I REALLY WROTE THAT? :p
+                    REQUEST_SIZE,
```
Comment (dadoonet) [inline src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java]: That's just crazy I left that around... Thanks a lot!

File: src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java [hunk 3]
```diff
@@ -383,8 +385,8 @@ private void addFilesRecursively(FileAbstractor path, String filepath, Instant l
 
         // TODO Optimize it. We can probably use a search for a big array of filenames instead of
         // Searching fo 50000 files (which is somehow limited).
-        private Collection<String> getFileDirectory(String path)
-                throws Exception {
+        // --> changed to 10000 which is the max. Higher values results to a default of 10 (pagination)
```
Comment (dadoonet) [inline src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java]: That's super strange as I would have expect that elasticsearch actually fail in such case.
I need to check elasticsearch code base to see if not failing is intentional.

File: src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java [hunk 4]
```diff
@@ -397,35 +399,47 @@ private void addFilesRecursively(FileAbstractor path, String filepath, Instant l
...
             return files;
+        }
+
+        private String getName(Object nameObject) {
+
+            if(nameObject instanceof List){
+                return String.valueOf (((List) nameObject).get(0));
+            }
 
+            throw new RuntimeException("search result, file.name not of type List<String>");
```
Comment (dadoonet) [inline src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java]: May be use here `FILE_FILENAME` instead of `file.name`?

File: src/main/java/fr/pilato/elasticsearch/crawler/fs/client/SearchResponse.java [hunk 5]
```diff
@@ -75,7 +75,7 @@ public String toString() {
 
         @Key("_source")
         private Map<String, Object> source;
-        @Key("_fields")
+        @Key("fields")
```
Comment (dadoonet) [inline src/main/java/fr/pilato/elasticsearch/crawler/fs/client/SearchResponse.java]: Nice catch

**Comment by dadoonet** [comment]
Thanks a lot for this!

I left some comments.

File: src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java (hunk 4 above)
Comment (rfelker) [inline src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java]: yes, you are right, that's wrong.

File: src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java (hunk 3 above)
Comment (rfelker) [inline src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java]: I'm curious about the result... :)

File: src/main/java/fr/pilato/elasticsearch/crawler/fs/client/SearchRequest.java (hunk 1 above)
Comment (rfelker) [inline src/main/java/fr/pilato/elasticsearch/crawler/fs/client/SearchRequest.java]: 👍

**Comment by dadoonet** [comment]
Thanks a lot @rfelker. As I rewrote a lot of stuff, I used a lot of your fixes in 620f252.
